deep_groove_project/src/main.py

```diff
@@ -1,53 +1,106 @@
 import torch
-import altair as alt
-import matplotlib.pyplot as plt
 import pandas as pd
-
+from torch import nn
+from torch.utils.data import DataLoader
+from src.FeatureDataset import FeatureDataset
+from src.NeuralNetwork import NeuralNetwork
+import matplotlib.pyplot as plt
 
 def main():
-    df = pd.read_csv("../data/first_dataset.csv")
-    show_standard_info(df)
-    show_correlation_grafs(df)
-    show_3d_scatter(df)
-
-
-def show_standard_info(df):
-    print("\n General Info:")
-    df.info()
-    print("\n Table:")
-    print(df.head())
-
-
-def show_correlation_grafs(df):
-    fields = ['Fr','n','Lifetime']
-
-    chart = alt.Chart(df).mark_point().encode(
-        alt.X(alt.repeat("column"), type='quantitative'),
-        alt.Y(alt.repeat("row"), type='quantitative')
-    ).properties(
-        width=200,
-        height=200
-    ).repeat(
-        row=fields,
-        column=fields[::1]
-    ).interactive()
-    chart.show()
-
-
-def show_3d_scatter(df):
-    fig = plt.figure()
-    ax = fig.add_subplot(projection='3d')
-
-    fr = df.iloc[:, 0]
-    n = df.iloc[:, 1]
-    lifetime = df.iloc[:, 2]
-
-    ax.scatter(fr, n, lifetime)
-    ax.set_xlabel('Fr')
-    ax.set_ylabel('n')
-    ax.set_zlabel('Lifetime')
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model = NeuralNetwork().to(device)
+    print(model)
+
+    #  Loading and preparing Dataframe
+    dataframe = pd.read_csv("../data/first_dataset.csv")
```

Comment:
In Contrast to your tutorial. This dataset could be committed. As long, as we do not use another solution, it is fine to commit the datasets which are unique to your project.